app/assets/javascripts/templates/admin/modals/business_address_info.html.haml

```diff
@@ -0,0 +1,7 @@
+%div
+  .margin-bottom-30
+    %p
+      {{ 'js.admin.modals.business_address_info.message' | t }}
+
+  .text-center
+    %input.button.red.icon-plus{ type: 'button', value: t('js.admin.modals.got_it'), ng: { click: 'close()' } }
```

Comment:
I might be wrong but I think that this call to `t` calls the Ruby method `I18n.t` when this template is compiled during deploy, via `rake assets:precompile`. The compilation just uses the default language and then the translated string is fixed and can't be changed by the user selecting another language. This happens only in staging and production, not development.

So I think that you need to use the AngularJS filter `t` as above or call the Javascript method `I18n.t`.

Reply:
@mkllnk I really just followed how we did the tooltip messages for tag_rule for this one. If you think it's better to go with the Javascript method we can do that, but then we need to consider changing the older tooltip messages as well because of the aforementioned translation issues on compilation on staging and production servers?

Reply:
Yes. I just tested this by precompiling assets locally and the "Got it" button is not translated.